### PR TITLE
o365azure rules - Work In Progress

### PR DESCRIPTION
**Pull request by juju4**
This is ongoing work
mostly rules from 
- https://0x00sec.org/t/a-blue-team-guide-to-azure-office-365-monitoring/14411
- https://expel.io/blog/seven-ways-to-spot-business-email-compromise-office-365/
Thanks a lot to them for sharing

Few issues
- count(var1, var2) is not supported by sigma currently
- arbitrary count threshold else sigmac failing
- how to make comparison between 2 variables like MailboxOwnerUPN != user
- a variable to automatically replace organization.tld by one or multiple values would be nice
- no option to handle geolocation

To be continued.

### CODE REVIEW

File: rules/o365azure/azuread_highfail_mfa_challenges.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    product: o365
+falsepositives:
+    - Penetration Test
+    - Buggy products
+    - Unknown
+level: medium
+detection:
+    selection:
+        LogonError: UserStrongAuthClientAuthNRequiredInterrupt
+    condition: selection | count(user) by clientip > 10
```
Comment (Neo23x0) [inline rules/o365azure/azuread_highfail_mfa_challenges.yml]: @thomaspatzke that's splunk syntax, isn't it? We do not support that yet, right?

File: rules/o365azure/azuread_locked_accounts.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    - Penetration Test
+    - Unknown
+level: medium
+detection:
+    selection:
+      Workload: AzureActiveDirectory
+      Operation: UserLoginFailed
+      LogonError: IdsLocked "Actor{}.ID"!="Unknown"
+# iplocation clientip
+    condition: selection | count(userid) > 5
```
Comment (Neo23x0) [inline rules/o365azure/azuread_locked_accounts.yml]: @thomaspatzke : Splunk syntax ...

File: rules/o365azure/o365_account_lockouts.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    - Unknown
+level: medium
+detection:
+    selection:
+      Operation: UserLoginFailed
+      LogonError: IdsLocked
+# filter? unique
+# filter? not existing in current blacklist 
+# iplocation clientip
+    condition: selection | count(clientip) > 5
```
Comment (Neo23x0) [inline rules/o365azure/o365_account_lockouts.yml]: @thomaspatzke : splunk syntax

File: rules/o365azure/o365_email_forwarding_external.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+# Forwarding to an external recipient by a user
+detection:
+    selection:
+        Workload: Exchange
+        Operation: "UpdateInboxRules" 
+    keywords:
+        - "Forward"
+#    condition: selection and keywords | count(user,dest,operation,ruleactions)
+#    condition: selection and keywords | count(user)
+    condition: selection and keywords | count(user) > 3
```
Comment (Neo23x0) [inline rules/o365azure/o365_email_forwarding_external.yml]: @thomaspatzke : splunk syntax

**Comment by Neo23x0** [comment]
Pull requests of that size take hours to process. I guess we have to exclude some of the rules and could merge others. Indentation should be consistent.

**Comment by juju4** [comment]
Thanks @Neo23x0 for review and Sorry that I let that sit since long.
Yes, upstream source was Splunk-based.

**Pull request closed without merging**